libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/causalWhatIfUtils.ts

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  Cohort,
+  ColumnCategories,
+  ICausalWhatIfData,
+  IDataset,
+  ifEnableLargeData,
+  JointDataset
+} from "@responsible-ai/core-ui";
+import _ from "lodash";
+
+export async function getTreatmentValue(
+  dataset: IDataset,
+  cohort: Cohort,
+  selectedIndex: number,
+  featureName: string,
+  setTestDataRow: (testDataRow: any) => void,
+  absoluteIndex?: number,
+  requestTestDataRow?: (
+    absoluteIndex: number,
+    abortSignal: AbortSignal
+  ) => Promise<any>
+): Promise<number> {
+  const featureKey =
+    JointDataset.DataLabelRoot + dataset.feature_names.indexOf(featureName);
+  let treatmentValue = cohort.getRow(selectedIndex)[featureKey];
+
+  if (ifEnableLargeData(dataset)) {
+    const testDataRow = await getTestDataRow(
```

Comment:
nit: for naming, why do we call it testDataRow? It is a bit confusing to me :P

Reply:
test_data_row is a backend api to get the test row using absolute index. I named the method with same name as api to avoid confusion.